packages/push/src/push_vm/push_state.rs

```diff
@@ -128,6 +133,27 @@ impl State for PushState {
     }
 }
 
+impl HasStdout for PushState {
+    type Stdout = Cursor<Vec<u8>>;
+
+    fn stdout(&mut self) -> &mut Self::Stdout {
+        &mut self.stdout
+    }
+
+    // TODO: Can we avoid the `clone()` here? It's not a huge deal (we'll
+    //   only call it once at the end of running the program), but it would
+    //   still be nice to avoid it if possible.
```

Comment:
No we can't, since the String later re-uses the cloned vec as it's internal buffer, which shouldn't depend on the current internal buffer (or can't)

Reply:
Gotcha, thanks for clarifying that.